sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -33,90 +33,177 @@
+          return mvr.lane_change_maneuver.end_speed;
+      case carma_planning_msgs::msg::Maneuver::INTERSECTION_TRANSIT_STRAIGHT:
+          return mvr.intersection_transit_straight_maneuver.end_speed;
+      case carma_planning_msgs::msg::Maneuver::INTERSECTION_TRANSIT_LEFT_TURN:
+          return mvr.intersection_transit_left_turn_maneuver.end_speed;
+      case carma_planning_msgs::msg::Maneuver::INTERSECTION_TRANSIT_RIGHT_TURN:
+          return mvr.intersection_transit_right_turn_maneuver.end_speed;
+      case carma_planning_msgs::msg::Maneuver::STOP_AND_WAIT:
+          return 0;
+      default:
+          RCLCPP_ERROR_STREAM(rclcpp::get_logger("sci_strategic_plugin"), "Requested end speed from unsupported maneuver type");
+          return 0;
+    }
+  }
+} // namespace anonymous
+
+SCIStrategicPlugin::SCIStrategicPlugin(const rclcpp::NodeOptions &options)
+  : carma_guidance_plugins::StrategicPlugin(options), config_(SCIStrategicPluginConfig())
 {
-  plugin_discovery_msg_.name = config_.strategic_plugin_name;
-  plugin_discovery_msg_.version_id = "v1.0";
-  plugin_discovery_msg_.available = true;
-  plugin_discovery_msg_.activated = true;
-  plugin_discovery_msg_.type = cav_msgs::Plugin::STRATEGIC;
-  plugin_discovery_msg_.capability = "strategic_plan/plan_maneuvers";
+  // Declare parameters
+  config_.vehicle_decel_limit_multiplier = declare_parameter<double>("vehicle_decel_limit_multiplier",   config_.vehicle_decel_limit_multiplier);
+  config_.vehicle_accel_limit_multiplier = declare_parameter<double>("vehicle_accel_limit_multiplier",   config_.vehicle_accel_limit_multiplier);
+  config_.stop_line_buffer = declare_parameter<double>("stop_line_buffer",   config_.stop_line_buffer);
+  config_.delta_t = declare_parameter<double>("delta_t",   config_.delta_t);
+  config_.min_gap = declare_parameter<double>("min_gap",   config_.min_gap);
+  config_.reaction_time = declare_parameter<double>("reaction_time",   config_.reaction_time);
+  config_.strategic_plugin_name = declare_parameter<std::string>("strategic_plugin_name",            config_.strategic_plugin_name);
+  config_.lane_following_plugin_name = declare_parameter<std::string>("lane_following_plugin_name",       config_.lane_following_plugin_name);
+  config_.intersection_transit_plugin_name = declare_parameter<std::string>("intersection_transit_plugin_name", config_.intersection_transit_plugin_name);
+  config_.vehicle_id = declare_parameter<std::string>("vehicle_id", config_.vehicle_id);
+  config_.veh_length = declare_parameter<double>("vehicle_length", config_.veh_length);
+  config_.vehicle_decel_limit = declare_parameter<double>("vehicle_deceleration_limit", config_.vehicle_decel_limit);
+  config_.vehicle_accel_limit = declare_parameter<double>("vehicle_acceleration_limit", config_.vehicle_accel_limit);
+
 };
 
-cav_msgs::Plugin SCIStrategicPlugin::getDiscoveryMsg() const
+carma_ros2_utils::CallbackReturn SCIStrategicPlugin::on_configure_plugin()
```

Comment:
Can we add the parameter update callback as well for the parameters not coming from vehicle configuration?

Reply:
added